agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -35,6 +40,27 @@
  */
 public final class RestrictorFactory {
 
+    private static final Pattern URL_PATTERN = Pattern.compile("(http[s]?://)((.+):(.+)@)?(.+)");
```

Comment:
I wonder whether, how to deal with 'user:' and ':pass'. I guess 'user:' is valid, when we use an empty password ? but ':pass' is probably really an error. 

So maybe we should allow `(?<protocol>http[s]?://)(?:(?<user>.+:)(?<password>.+)?@)?.*` (here with named capturing groups), so that empty passwords are also matched ?

BTW, what is with password containing `@` ? (don't know whether that is supported). however should work here since the regexp is greedy. Maybe you could add a test with an password containing a `@`, too ?